webrtcclient.js

```diff
@@ -196,4 +196,33 @@ WebRTCClient.prototype.getStats = function() {
   });
 };
 
+WebRTCClient.prototype.waitForAudio(silence) {
+  return this.driver.executeAsyncScript(function(constraints) {
+    var callback = arguments[arguments.length - 1];
+
+    var remoteStream = pc.getRemoteStreams()[0];
+
+    var ac = new AudioContext();
+    var analyser = ac.createAnalyser();
+    analyser.fftSize = 128;
+    analyser.smoothingTimeConstant = 0;
+    var fftBins = new Uint8Array(analyser.frequencyBinCount);
+    var sourceNode = ac.createMediaStreamSource(new MediaStream([remoteStream.getAudioTracks()[0]]));
+    sourceNode.connect(analyser);
+    function poll() {
+      analyser.getByteFrequencyData(fftBins);
+      var sum = fftBins.reduce((a, b) => a + b);
+      if ((silence && sum === 0) || (!silence && sum > 0)) {
+        ac.close()
+        .then(() => {
+          callback();
+        });
+      } else {
+        setTimeout(poll, 100);
```

Comment:
I would consider running this more often. Perhaps even in `requestAnimationFrame` unless your machine is an android emulator or some such. The fftSize is small which also means you're not looking at that much temporal data each `getBytFrequencyData()`.

Depends on what you are analysing though. A tone is fine, but pulses could be .. problematic with some bad luck.